### ISSUE TITLE
DeserializeLambda: Type not supported

### ISSUE THREAD

**Issue by fjoppe** [intellifactory/websharper#524]
Hi,
I am trying to do something with Websocket, and for this I need Websharper.Json, and especially the serialize/deserialize functions.

I have an akka.net actor-system which orchestrates state, and this system has some messages. I wanted to reuse these messages, to be send through the Websocket, and therefore I tried to make them Websharper.Json ready with attribute `[<NamedUnionCases>]`.

So I have two aggregate DU's, [the source is here](https://github.com/fjoppe/ThreeIsACrowdActorBased/blob/master/GameEngine.Common/Messages.fs). "PlayerMessage" is send from client to actor system. "PlayerMessageResponse" is send from actor system to client.

So for, so good.

One of the cases in "PlayerMessageResponse" gives problems. When I compile (not the library, but the ws application using the library) gives the following compilation error:

> Error		GameEngine.Common.PlayerMessageResponse: DeserializeLambda: Type not supported: GameEngine.Common.TileType	GameUI.Suave	F:\\....\WebSharper.3.6.7.227\build\WebSharper.targets	60

Detail, the library containing "PlayerMessageResponse" was first an ordinary library, but I added a property to the project as explained [here](http://stackoverflow.com/a/32844308/4653407). This fixed the previous problem that no DU was compiling at all. So now its reduced to "TileType".

The source of [TileType is found here](https://github.com/fjoppe/ThreeIsACrowdActorBased/blob/master/GameEngine.Common/TileType.fs). Decorated with an attribute or not; it does not compile. 

My suspicion is that this DU is different because it actualy is an enumeration and I assign constants to it. Still, I'm stuck here, so I suspect a defect.

**Comment by Tarmil** [comment]
Yes this seems to be an issue with enumerations, I'm investigating.

**Comment by fjoppe** [comment]
Hi,

I just upgraded to WS 3.6.10.230. I don't know what to expect, but this issue is still in there.

**Comment by granicz** [comment]
That's no good news, reopening ticket then for Loic to double-check.

**Comment by Tarmil** [comment]
This is strange. I just added [test cases for enum types](https://github.com/intellifactory/websharper/blob/master/tests/WebSharper.Web.Tests/ClientSideJson.fs#L280-L290) and they are passing just fine.

**Comment by fjoppe** [comment]
I still see the issue. However I have a suspect. I get this error in a [Websharper.Suave (3.6.4.30) project](https://github.com/fjoppe/ThreeIsACrowdActorBased/tree/master/GameUI.Suave), but a [Websharper C/S project](https://github.com/fjoppe/ThreeIsACrowdActorBased/tree/master/GameUI) which includes the same dll, does not provide this error. 

The WS.Suave project gets a list of other packages from nuget, including Fsharp.Core 3.1.2.5. In coïncidence with a propriaty project at work this week, I discovered that different Fsharp.Core's can influence how WS works. I found that Fsharp.Data and WS, in sepearte dll-projects, were not working together very well, until I used Fsharp.Core 4.4.0.0 in both projects. 

So the Fsharp.Core is my prime suspect. As I'm turning away from the idea of using a Websocket in a Websharper.Suave project, [due to complications](http://websharper.com/question/81214/running-a-ws-websocket-with-suave), maybe its good to reconsider if you want to fix this, for this type of situation.
